### PR TITLE
EMPs now randomize suit sensors if they're on

### PR DESCRIPTION
**Pull request by Couls**
## About The Pull Request
Makes suit sensors react to EMPs

## Why It's Good For The Game
It's just a cool feature and since jumpsuits can have their sensors broken by damage I don't see why they wouldn't be impacted by EMPs if they were on

## Changelog
:cl:
add: suit sensors are now randomized when caught in an EMP
/:cl:

### CODE REVIEW

**Comment by Shadowflame909** [comment]
should turn them on if they're off.

**Comment by Couls** [comment]
That would be easy to change but I didn’t really wanna mess with balance all that much because that’ll buff EMPs

**Comment by wesoda25** [comment]
Randomizing is perfect, theres a chance for your victim to go off but you go on

**Comment by Time-Green** [comment]
New medical doctor meta

**Comment by Akrilla** [comment]
Rough CMO turning up those suit sensors

**Comment by Shadowflame909** [comment]
I think emps turning suit sensors on would be a nerf. Make people think twice before revealing their locations.

Also, this won't help CMOS other gamers. It only affects suit sensors when ON. So it'll just make more people with suit sensors off.

Add that nerf Couls. The balance must be restored. Since this is also a buff.

**Comment by Couls** [comment]
It would make no sense for an EMP to turn your sensors on.. thats like an EMP turning your cellphone on when it was off

File: code/modules/clothing/under/_under.dm [hunk 1]
```diff
@@ -52,6 +52,14 @@
 		//make the sensor mode favor higher levels, except coords.
 		sensor_mode = pick(SENSOR_OFF, SENSOR_LIVING, SENSOR_LIVING, SENSOR_VITALS, SENSOR_VITALS, SENSOR_VITALS, SENSOR_COORDS, SENSOR_COORDS)
 
+/obj/item/clothing/under/emp_act()
+	if(has_sensor > NO_SENSORS)
+		sensor_mode = pick(SENSOR_OFF, SENSOR_OFF, SENSOR_OFF, SENSOR_LIVING, SENSOR_LIVING, SENSOR_VITALS, SENSOR_VITALS, SENSOR_COORDS)
+		if(ismob(loc))
+			var/mob/M = loc
+			to_chat(M,"<span class='warning'>The sensors on the [src] change rapidly!</span>")
+	..()
```
Comment (ninjanomnom) [inline code/modules/clothing/under/_under.dm]: Use `. = ..()` at the top

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**